python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -620,3 +621,133 @@ def get_lsst_magnitudes(self):
+            # it's "dry run" to determine what columns are required from
+            # the database
+            return numpy.zeros((len(bandpassDict.keys()),0))
+
+        magListOut = []
+
+        for sedName, magNorm in zip(sedNameList, magNormList):
+            magTag = bandpassTag+'_'+sedName
+            if sedName not in self._ssmMagNormDict or magTag not in self._ssmMagDict:
+                dummySed = Sed()
+                dummySed.readSED_flambda(os.path.join(self._file_dir, self._spec_map[sedName]))
+                fnorm = dummySed.calcFluxNorm(magNorm, self._normalizing_bandpass)
+                dummySed.multiplyFluxNorm(fnorm)
+                magList = bandpassDict.magListForSed(dummySed)
+                self._ssmMagDict[magTag] = magList
+                self._ssmMagNormDict[sedName] = magNorm
+            else:
+                dmag = magNorm - self._ssmMagNormDict[sedName]
+                magList = self._ssmMagDict[magTag] + dmag
+            magListOut.append(magList)
+
+        return numpy.array(magListOut).transpose()
+
+
+    @compound('lsst_u','lsst_g','lsst_r','lsst_i','lsst_z','lsst_y')
+    def get_lsst_magnitudes(self):
+        """
+        getter for LSST magnitudes of solar system objects
+        """
+
+        if not hasattr(self, 'lsstBandpassDict'):
+            self.lsstBandpassDict = BandpassDict.loadTotalBandpassesFromFiles()
+
+        return self._magnitudeGetter(self.lsstBandpassDict)
+
+
+    @compound('dmagTrailing', 'dmagDetection')
+    def get_ssm_dmag(self):
+        """
+        This getter will calculate:
+
+        dmagTrailing: the offset in m5 used to represent the loss in signal to noise
+        resulting from the fact that the object's motion smears out its PSF
+
+        dmagDetection: the offset in m5 used to represent the shift in detection
+        threshold resulting from the fact that the object's motion smears out
+        its PSF
+        """
+
+        if self.obs_metadata.seeing is None:
+            raise RuntimeError("Cannot calculate dmagTraling/dmagDetection. "
+                               "Your catalog's ObservationMetaData does not "
+                               "specify seeing.")
+
+        if not isinstance(self.obs_metadata.bandpass, str):
+            raise RuntimeError("dmagTrailing/dmagDetection calculation is confused. "
+                               "Your catalog's ObservationMetaData contains multiple "
+                               "bandpasses.  It is unclear what seeing value you mean. "
+                               "Re-create your catalog with only one bandpass.")
+
+        if not hasattr(self, 'photParams') or self.photParams is None:
+            raise RuntimeError("You cannot calculate dmagTrailing/dmagDetection. "
+                               "Your catalog does not have an associated PhotometricParameters "
+                               "member variable.  It is impossible to know what the exposure time is.")
```

Comment:
Partially it was the (mistaken) belief that exposure time is a fixed constant for the survey.  Partially it is what they are used for.

PhotometricParameters was originally designed to hold parameters that the signal to noise methods in sims_photUtils needed.  ObservationMetaData was designed (originally) to hold information about the pointing that InstanceCatalog needed when querying fatboy.  Obviously, things have grown since then.  Anyway: the signal to noise methods require seeing and exposure time.  Ergo, seeing and exposure time went in PhotometricParameters.

If we were to move to a more physically-motivated system (hardware/site parameters going in PhotometricParameters; pointing parameters going in ObservationMetaData), we would be talking about a system where the signal to noise methods in sims_photUtils take both an ObservationMetaData and a PhotometricParameters (though, I guess even my proposed solution to this problem ends in that happening).